v2/pkg/protocols/common/variables/variables.go

```diff
@@ -8,11 +8,13 @@ import (
 	"github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/expressions"
 	"github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/generators"
 	"github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/interactsh"
-	"github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/marker"
 	"github.com/projectdiscovery/nuclei/v2/pkg/types"
 	"github.com/projectdiscovery/nuclei/v2/pkg/utils"
+	"github.com/projectdiscovery/stringsutil"
 )
 
+var DefaultProtocolVariables = []string{"FQDN", "RDN", "DN", "TLD", "SD", "BaseURL", "RootURL", "Hostname", "Host", "Port", "Path", "File", "Scheme", "Input"}
```

Comment:
@ShubhamRasal I think we can revisit lazyVar implementation . 
Keeping track of whitelist does not seem like best solution . we can try some other method if possible 

cc: @Mzack9999

Reply:
A few alternatives:
- static map/helper function in each protocol with known variable names
- As noted by @tarunKoyalwar, the to map shared conversion functions (http, dns) are in the utils package (as they are used by multiple tools). Not sure if centralizing the logic might be an option